Tests/SonarScanner.MSBuild.Shim.Test/SonarScannerWrapperTests.cs

```diff
@@ -317,6 +317,41 @@ public void SonarScanner_WhenSettingJavaHomePathFails_AWarningIsLogged(string pa
         logger.DebugMessages.Should().BeEmpty();
     }
 
+    [TestMethod]
+    public void SonarScanner_ScannerOptsSettingSonarScannerOptsEmpty()
+    {
+        var logger = new TestLogger();
+        var mockRunner = new MockProcessRunner(executeResult: true);
+        var config = new AnalysisConfig();
+        config.ScannerOptsSettings.Add(new Property("sonar.scanner.opts", "value"));
+
+        using (new EnvironmentVariableScope())
+        {
+            var result = ExecuteJavaRunnerIgnoringAsserts(config, [], logger, "exe file path", "properties file path", mockRunner);
+            result.Should().BeTrue();
+        }
+
+        CheckEnvVarExists("SONAR_SCANNER_OPTS", "-Dsonar.scanner.opts=value", mockRunner);
+    }
+
+    [TestMethod]
+    public void SonarScanner_ScannerOptsSettingSonarScannerOptsNotEmpty()
+    {
+        var logger = new TestLogger();
+        var mockRunner = new MockProcessRunner(executeResult: true);
+        var config = new AnalysisConfig();
+        config.ScannerOptsSettings.Add(new Property("sonar.scanner.opts", "value"));
+
+        using (var scope = new EnvironmentVariableScope())
+        {
+            scope.SetVariable("SONAR_SCANNER_OPTS", "-Dsonar.anything.config=existing");
+            var result = ExecuteJavaRunnerIgnoringAsserts(config, [], logger, "exe file path", "properties file path", mockRunner);
+            result.Should().BeTrue();
+        }
+
+        CheckEnvVarExists("SONAR_SCANNER_OPTS", "-Dsonar.anything.config=existing -Dsonar.scanner.opts=value", mockRunner);
+    }
+
```

Comment:
Add a test where the existing `scope.SetVariable("SONAR_SCANNER_OPTS", "-Dsonar.anything.config=existing");` is the same as `config.ScannerOptsSettings.Add(new Property("sonar.anything.config", "new"));`